scripts/linux-prereqs.sh

```diff
@@ -22,321 +30,215 @@ if [ "$1" = "false" ]; then NETCOREDEPS=0; else NETCOREDEPS=1; fi
 if [ "$2" = "false" ]; then KEYRINGDEPS=0; else KEYRINGDEPS=1; fi
 if [ "$3" = "false" ]; then BROWSERDEPS=0; else BROWSERDEPS=1; fi
 
-# If not already root, validate user has sudo access and error if not.
-if [ $(id -u) -ne 0 ]; then
-    echo "To begin the installation process, your OS will now ask you to enter your"
-    echo "admin / root (sudo) password."
-    echo ""
-    # Validate user actually can use sudo
-    sudo -v > /dev/null 2>&1
-    if [ $? -ne 0 ]; then
-        echo ""
-        echo "(!) Dependency installation failed! You do not have the needed admin / root"
-        echo "    access to install Live Share's dependencies. Contact your system admin"
-        echo "    and ask them to install the required libraries described here:"
-        echo "    https://aka.ms/vsls-docs/linux-required-lib-details"
-        echo ""
-        echo "Press enter to dismiss this message."
-        read
-        exit 3
-    fi
-fi
+# Disable history substitution given use of "!" or errors occur in certain distros
+set H+
+
+# Utility function for exiting
+exitScript()
+{
+    echo -e "\nPress enter to dismiss this message"
+    read
+    exit $1
+}
 
 # Wrapper function to only use sudo if not already root
-sudoif()
+sudoIf()
 {
-    if [ $(id -u) -ne 0 ]; then
+    if [ "$(id -u)" -ne 0 ]; then
         set -- command sudo "$@"
     fi
     "$@"
 }
 
-#openSUSE - Has to be first as apt is aliased to zypper
-if type zypper > /dev/null 2>&1; then
-    echo "(*) Detected SUSE (unoffically/community supported)"
-    echo ""
+# Utility function that waits for any existing installation operations to complete
+# on Debian/Ubuntu based distributions and then calls apt-get
+aptSudoIf() 
+{
+    while sudoIf fuser /var/lib/dpkg/lock >/dev/null 2>&1; do
+        echo -ne "(*) Waiting for other package operations to complete.\r"
+        sleep 0.2
+        echo -ne "(*) Waiting for other package operations to complete..\r"
+        sleep 0.2
+        echo -ne "(*) Waiting for other package operations to complete...\r"
+        sleep 0.2
+        echo -ne "\r\033[K"
+    done
+    sudoIf apt-get "$@"
+}
 
+# Installs .NET Core dependencies if not disabled
+checkNetCoreDeps(){
     if [ $NETCOREDEPS -ne 0 ]; then
+        echo -e "\n(*) Verifying .NET Core dependencies..."
         # Install .NET Core dependencies
-        sudoif zypper -n in libopenssl1_0_0 libicu krb5 libz1
-        if [ $? -ne 0 ]; then
-            echo "(!) .NET Core dependency install failed! Press enter to dismiss this message."
-            read
-            exit 1
+        if ! "$@"; then
```

Comment:
There is a particular reason why we are using $@ and passing the installation payload as multiple parameter and not $1 and passing the payload as single parameter?

Reply:
I need to run a different function depending on the distro.  For Debian/Ubuntu it uses aptSudoIf, others it uses sudoIf. Passing that in as a string does not work.

Reply:
What about using sudoapt/sudoif as $1 and the package manager payload as $2?

Reply:
Sure - Adapted code to that model instead.